postprocessing/processors/ReductionProcessor.py

```diff
@@ -0,0 +1,119 @@
+from .base_processor import BaseProcessor
+from . import job_handling
+
+import json
+import logging
+import os
+import subprocess
+import sys
+
+
+class ReductionProcessor(BaseProcessor):
+    _message_queue = "/queue/REDUCTION.DATA_READY"
+
+    STARTED_QUEUE = "/queue/REDUCTION.STARTED"
+    COMPLETED_QUEUE = "/queue/REDUCTION.COMPLETE"
+    ERROR_QUEUE = "/queue/REDUCTION.ERROR"
+    DISABLED_QUEUE = "/queue/REDUCTION.DISABLED"
+
+    def __init__(self, data, conf, send_function):
+        """
+        Initialize the processor
+        @param data: data dictionary from the incoming message
+        @param conf: configuration object
+        @param send_function: function to call to send AMQ messages
+        """
+        super(ReductionProcessor, self).__init__(data, conf, send_function)
+
+    def __call__(self):
+        """
+        Reduction process using job submission.
+        """
+
+        try:
+            self.send(ReductionProcessor.STARTED_QUEUE, json.dumps(self.data))
+            # get instrument shared directory
+            instrument_shared_dir = os.path.join(
+                "/", self.facility, self.instrument, "shared", "autoreduce"
+            )
+            if len(self.configuration.dev_instrument_shared) > 0:
+                instrument_shared_dir = self.configuration.dev_instrument_shared
+
+            # get the proposal shared directory
+            proposal_shared_dir = os.path.join(
+                "/",
+                self.facility,
+                self.instrument,
+                self.proposal,
+                "shared",
+                "autoreduce",
+            )
+            # Allow for an alternate output directory, if defined
+            if len(self.configuration.dev_output_dir) > 0:
+                proposal_shared_dir = self.configuration.dev_output_dir
+            logging.info("Using output directory: %s" % proposal_shared_dir)
+
+            # Set logging directory
+            log_dir = os.path.join(proposal_shared_dir, "reduction_log")
+            if not os.path.exists(log_dir):
+                os.makedirs(log_dir)
+
+            # Look for run summary script
+            summary_script = os.path.join(
+                instrument_shared_dir, "sumRun_%s.py" % self.instrument
+            )
+            if os.path.exists(summary_script) is True:
+                summary_output = os.path.join(
+                    proposal_shared_dir,
+                    "%s_%s_runsummary.csv" % (self.instrument, self.proposal),
+                )
+                cmd = (
+                    "python "
```

Comment:
On rhel9 (and the conda environments), python -> python3 so changing that part of the command isn't necessary.

Reply:
Oh good! I had misunderstood. I will resolve this comment.

Reply:
On rhel9 the package `python-unversioned-command` must be installed so that `python` points to `python3`, otherwise `python` doesn't exist.

Unless we make that a dependency in the postprocessing rpm we can't guarantee that it is installed.